frontend/web/components/pages/EnvironmentSettingsPage.tsx

```diff
@@ -0,0 +1,886 @@
+import AccountStore from 'common/stores/account-store'
+import { Link, RouterChildContext } from 'react-router-dom'
+import { enableFeatureVersioning } from 'common/services/useEnableFeatureVersioning'
+import AddMetadataToEntity from 'components/metadata/AddMetadataToEntity'
+import { getSupportedContentType } from 'common/services/useSupportedContentType'
+import EnvironmentVersioningListener from 'components/EnvironmentVersioningListener'
+import Format from 'common/utils/format'
+import Setting from 'components/Setting'
+import API from 'project/api'
+import AppActions from 'common/dispatcher/app-actions'
+import { Environment } from 'common/types/responses'
+import PanelSearch from 'components/PanelSearch'
+import moment from 'moment'
+import Panel from 'components/base/grid/Panel'
+import ProjectProvider from 'common/providers/ProjectProvider'
+import InputGroup from 'components/base/forms/InputGroup'
+import Utils from 'common/utils/utils'
+import { Webhook } from 'common/types/webhooks'
+import useFormNotSavedModal from 'components/hooks/useFormNotSavedModal'
+
+const showDisabledFlagOptions: { label: string, value: boolean | null }[] = [
+  { label: 'Inherit from Project', value: null },
+  { label: 'Disabled', value: false },
+  { label: 'Enabled', value: true },
+]
+
+interface EnvironmentSettingsPageProps {
+  // Router props
+  match: {
+    params: {
+      projectId: string
+      environmentId: string
+    }
+  }
+  router: RouterChildContext['router']
+  // Webhook props from HOC
+  webhooks: Webhook[]
+  webhooksLoading: boolean
+  getWebhooks: () => void
+  createWebhook: (webhook: Partial<Webhook>) => Promise<void>
+  saveWebhook: (webhook: Webhook) => Promise<void>
+  deleteWebhook: (webhook: Webhook) => Promise<void>
+}
+
+const EnvironmentSettingsPage: React.FC<EnvironmentSettingsPageProps> = ({ createWebhook, deleteWebhook, getWebhooks, match, router, saveWebhook, webhooks, webhooksLoading }) => {
+  const store = getStore()
+  const [currentEnv, setCurrentEnv] = useState<Environment | null>(null)
+  const [roles, setRoles] = useState<any[]>([])
+  const [environmentContentType, setEnvironmentContentType] = useState<any>(null)
+
+  const has4EyesPermission = Utils.getPlansPermission('4_EYES')
+  const metadataEnable = Utils.getPlansPermission('METADATA')
+
+  const onDiscard = () => {
+    const env = ProjectStore?.getEnvs()?.find(
+      (env: Environment) => env.api_key === match.params.environmentId,
+    )
+    if (env && currentEnv) {
+      setCurrentEnv({ ...currentEnv, description: env.description, name: env.name })
+    }
+  }
+
+  const [DirtyFormModal, setIsDirty, isDirty] = useFormNotSavedModal(router.history, { onDiscard: onDiscard })
+
+  const getEnvironment = useCallback(async () => {
+    const env = ProjectStore?.getEnvs()?.find(
+      (env: Environment) => env.api_key === match.params.environmentId,
+    )
+    setCurrentEnv(env)
```

Comment:
This method is somehow being called in a loop every time `currentEnv` gets updated by setCurrentEnv(env).

Reply:
<img width="1178" alt="Screenshot 2025-02-12 at 16 10 59" src="https://github.com/user-attachments/assets/67fc8bde-a705-48ee-9a96-2b7d7622fdfa" />

Reply:
Thanks for the fix, let me know if anything else required in this PR